internal/services/servicefabric/service_fabric_cluster_resource_test.go

```diff
@@ -813,6 +813,10 @@ resource "azurerm_service_fabric_cluster" "test" {
     client_endpoint_port = 2020
     http_endpoint_port   = 80
   }
+
+  lifecycle {
+    ignore_changes = [cluster_code_version]
+  }
```

Comment:
For this to work we will also need to split to combined `CreateUpdate` method of the resource. Could you please do that as well?

Reply:
You mean we need to split create method and update method? May I ask why we need to do this? Seems it's not a simple change as there are many properties to work and test.

Reply:
@neil-yechenwei this is explained in the [best practices section of the contributor docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#separate-create-and-update-methods).

However, as mentioned in the guidelines I provided for the task of reviewing all Computed+Optional properties, it isn't necessary to remove _all_ Computed+Optional properties, just ones where there is a need for a user to reset/empty the property so that they can use API defaults. 

In this particular case I think we can answer the question of whether we need to remove `Computed` here by considering the following:
1. Is it a common use case to want to update the `upgrade_mode` from `Automatic` to `Manual` and unset the `cluster_code_version`?
2. Are there currently issues trying to do the above while `Computed` is set on `cluster_code_version`?

Could you please look into the two points above? Once we have clarity on that we will know whether removing `Computed` here is necessary or not.

Reply:
@stephybun ,

For your question 2, there is no issue.

For your question 1, I think users typically do not frequently modify the upgrade mode and cluster code version of Azure Service Fabric clusters. Here are some specific reasons:

Stability of Upgrade Mode: Azure Service Fabric provides two primary upgrade modes: automatic and manual. Generally, users select one mode based on their environmental needs and use it long-term.

Compatibility of Cluster Code Versions: Changing the cluster code version requires thorough planning and testing, as different versions may include new features, performance improvements, and bug fixes, but could also introduce incompatibilities or new issues. Frequent changes to the code version increase the risk of system instability. Therefore, users typically perform version upgrades only after sufficient testing and do not change versions frequently.

Operational Complexity and Risk: Each modification of the upgrade mode or code version involves configuration changes and system restarts, leading to service disruptions or system instability. To mitigate risks and reduce the impact on services, users tend to make these changes only when necessary, rather than frequently.

Operational Costs: Frequent changes to the upgrade mode or code version increase the complexity and cost of operations. The IT team needs to invest more time in testing, monitoring, and troubleshooting, which significantly consumes resources.

Reply:
Thanks for the clarification @neil-yechenwei. I think we can leave `Computed` here and skip this property. You can go ahead and close the PR 👍